### ISSUE TITLE
评估方式与评估结果

### ISSUE THREAD

**Issue by superlrt** [OpenBMB/DEBATER#1]
您好，我按照您给出的代码，对模型进行了训练和评估。但无论是我自己训练的模型，还是您给出的check point，都无法达到您论文中的评估效果，例如在scifact数据集下，得分只有0.59左右，无法达到您论文中给出的0.7。是我哪一个步骤出错了吗？还是我忽略了一些评估细节？

**Comment by SiberiaWolfPack** [comment]
@superlrt 您好，感谢您对工作的关注，请您提供更多的相关信息以便定位问题。

**Comment by superlrt** [comment]
您好，我运行您的llm2vec_models.py代码，无法运行。我自己写了一个评估代码如下，结果与论文相差很多。如果您的论文数据使用llm2vec_models.py运行得到的，可以告诉我环境的每个库的版本号吗？
`
from beir import util, LoggingHandler
from beir.retrieval import models
from beir.datasets.data_loader import GenericDataLoader
from beir.retrieval.evaluation import EvaluateRetrieval
from beir.retrieval.search.dense import DenseRetrievalExactSearch as DRES

import logging
import pathlib, os
import torch

from sentence_transformers import SentenceTransformer

logging.basicConfig(format='%(asctime)s - %(message)s',
                    datefmt='%Y-%m-%d %H:%M:%S',
                    level=logging.INFO,
                    handlers=[LoggingHandler()])

dataset = "nfcorpus"
url = f"https://public.ukp.informatik.tu-darmstadt.de/thakur/BEIR/datasets/{dataset}.zip"
out_dir = os.path.join(pathlib.Path(__file__).parent.absolute(), "datasets")
data_path = util.download_and_unzip(url, out_dir)

corpus, queries, qrels = GenericDataLoader(data_folder=data_path).load(split="test")

model = DRES(models.HuggingFace(
    model_path="/data/MiniCMP-2B",
    # peft_model_path="/data/test_models/DEBATE-2B",
    # peft_model_path="/data/test_models/RL-6/checkpoint-2400",
    max_length=512,
    pooling="eos",
    normalize=True,
    prompts={'query': 'Given a question, retrieve relevant documents that best answer the question: '}),
    batch_size=4)



retriever = EvaluateRetrieval(model, score_function="cos_sim") # or "dot" for dot product
results = retriever.retrieve(corpus, queries)

ndcg, _map, recall, precision = retriever.evaluate(qrels, results, retriever.k_values)

**Comment by SiberiaWolfPack** [comment]
@superlrt 
torch>=2.4.1
peft>=0.12.0
transformers>=4.43.1,<=4.44.2
Thank you for your continued attention. You should take a closer look at the code of llm2vec_models.py. I will update the version numbers of these packages in requirements.txt.

**Comment by howardatri** [comment]
请问你是怎么复现的？pytorch和cuda版本是？
